packages/typespec-python/dev_requirements.txt

```diff
@@ -1,2 +1,12 @@
--r ../../eng/requirements.txt
--r ../../eng/dev_requirements.txt
+pyright==1.1.389
+pylint==3.2.7
+tox==4.16.0
+mypy==1.13.0
+azure-pylint-guidelines-checker==0.0.8
+colorama==0.4.6
+debugpy==1.8.2
+pytest==8.3.2
+coverage==7.6.1
+black==24.8.0
+ptvsd==4.3.2
+types-PyYAML==6.0.12.8
```

Comment:
This file shall be synced with https://github.com/microsoft/typespec/blob/main/packages/http-client-python/generator/dev_requirements.txt